src/Evolution/Systems/GrMhd/GhValenciaDivClean/TimeDerivativeTerms.hpp

```diff
@@ -0,0 +1,212 @@
+          comoving_magnetic_field_one_form,
+      const tnsr::aa<DataVector, 3_st>& spacetime_metric,
+      const tnsr::aa<DataVector, 3_st>& /*pi*/,
+      const tnsr::iaa<DataVector, 3_st>& /*phi*/,
+      const Scalar<DataVector>& /*gamma0*/,
+      const Scalar<DataVector>& /*gamma1*/,
+      const Scalar<DataVector>& /*gamma2*/,
+      const tnsr::a<DataVector, 3_st>& /*gauge_function*/,
+      const tnsr::ab<DataVector, 3_st>& /*spacetime_deriv_gauge_function*/,
+      const TupleType& args) noexcept {
+    trace_reversed_stress_energy(
+        local_stress_energy, four_velocity_buffer_one_form,
+        comoving_magnetic_field_one_form,
+        tuples::get<
+            TemporaryReference<hydro::Tags::RestMassDensity<DataVector>>>(args),
+        tuples::get<
+            TemporaryReference<hydro::Tags::SpecificEnthalpy<DataVector>>>(
+            args),
+        tuples::get<TemporaryReference<
+            ValenciaTempTag<hydro::Tags::SpatialVelocityOneForm<
+                DataVector, 3_st, Frame::Inertial>>>>(args),
+        tuples::get<TemporaryReference<
+            ValenciaTempTag<hydro::Tags::MagneticFieldOneForm<
+                DataVector, 3_st, Frame::Inertial>>>>(args),
+        tuples::get<TemporaryReference<
+            ValenciaTempTag<hydro::Tags::MagneticFieldSquared<DataVector>>>>(
+            args),
+        tuples::get<TemporaryReference<ValenciaTempTag<
+            hydro::Tags::MagneticFieldDotSpatialVelocity<DataVector>>>>(args),
+        tuples::get<TemporaryReference<hydro::Tags::LorentzFactor<DataVector>>>(
+            args),
+        tuples::get<TemporaryReference<
+            ValenciaTempTag<grmhd::ValenciaDivClean::TimeDerivativeTerms::
+                                OneOverLorentzFactorSquared>>>(args),
+        tuples::get<TemporaryReference<hydro::Tags::Pressure<DataVector>>>(
+            args),
+        spacetime_metric,
+        tuples::get<TemporaryReference<
+            gr::Tags::Shift<3_st, Frame::Inertial, DataVector>>>(args),
+        tuples::get<TemporaryReference<gr::Tags::Lapse<DataVector>>>(args));
+  }
+};
+}  // namespace detail
+
+/*!
+ * \brief Compute the RHS terms and flux values for both the Generalized
+ * Harmonic formulation of Einstein's equations and the Valencia formulation of
+ * the GRMHD equations with divergence cleaning.
+ *
+ * \details The bulk of the computations in this class dispatch to
+ * `GeneralizedHarmonic::TimeDerivative` and
+ * `grmhd::ValenciaDivClean::TimeDerivativeTerms` as a 'product system' -- each
+ * independently operating on its own subset of the supplied variable
+ * collections.
+ * The additional step is taken to compute the trace-reversed stress energy
+ * tensor associated with the GRMHD part of the system and add its contribution
+ * to the \f$\partial_t \Pi_{a b}\f$ variable in the Generalized Harmonic
+ * system, which is the only explicit coupling required to back-react the effect
+ * of matter on the spacetime solution.
```

Comment:
Maybe add a `\note` at the end stating something like `The MHD calculation reuses any spacetime quantities in its argument_tags that are computed by the GH time derivative. However, other quantities that aren't computed by the GH time derivative like the extrinsic curvature are currently still grabbed from the DataBox. Those calculations can be explicitly inlined here to reduce memory pressure and the number of compute tags`?